src/sinks/datadog/logs/sink.rs

```diff
@@ -0,0 +1,398 @@
+        let mut members: Vec<BTreeMap<String, Value>> = Vec::with_capacity(batch.len());
+        let mut finalizers: EventFinalizers = EventFinalizers::default();
+        for mut event in batch.into_iter() {
+            {
+                let log = event.as_mut_log();
+                log.rename_key_flat(self.log_schema_message_key, "message");
+                log.rename_key_flat(self.log_schema_timestamp_key, "date");
+                log.rename_key_flat(self.log_schema_host_key, "host");
+                self.encoding.apply_rules(&mut event);
+            }
+
+            let (fields, mut metadata) = event.into_log().into_parts();
+            members.push(fields);
+            finalizers.merge(metadata.take_finalizers());
+        }
+        (members, finalizers)
+    }
+
+    fn build(
+        &self,
+        api_key: Arc<str>,
+        batch: Vec<Event>,
+    ) -> Result<LogApiRequest, RequestBuildError> {
+        let (members, finalizers) = self.dissect_batch(batch);
+
+        let total_members = members.len();
+        assert!(total_members <= MAX_PAYLOAD_ARRAY);
+        let body: Vec<u8> = serde_json::to_vec(&members).expect("failed to encode to json");
+        let serialized_payload_bytes_len = body.len();
+        if serialized_payload_bytes_len > MAX_PAYLOAD_BYTES {
+            return Err(RequestBuildError::PayloadTooBig);
+        }
+        metrics::histogram!(
+            "encoded_payload_size_bytes",
+            serialized_payload_bytes_len as f64
+        );
+        let (encoded_body, is_compressed) = match self.compression {
+            Compression::None => (body, false),
+            Compression::Gzip(level) => {
+                let level = level.unwrap_or(GZIP_FAST);
+                let mut encoder = GzEncoder::new(
+                    Vec::with_capacity(serialized_payload_bytes_len),
+                    flate2::Compression::new(level as u32),
+                );
+
+                encoder
+                    .write_all(&body)
+                    .map_err(|error| RequestBuildError::Io { error })?;
+                (encoder.finish().expect("failed to encode"), true)
+            }
+        };
+        Ok(LogApiRequest {
+            serialized_payload_bytes_len,
+            payload_members_len: total_members,
+            api_key: Arc::clone(&api_key),
+            is_compressed,
+            body: encoded_body,
+            finalizers,
+        })
+    }
+}
+
+#[async_trait]
+impl<S> StreamSink for LogSink<S>
+where
+    S: Service<LogApiRequest> + Send + 'static,
+    S::Future: Send + 'static,
+    S::Response: Response + Send + 'static,
+    S::Error: Debug + Into<crate::Error> + Send,
+{
+    async fn run(&mut self, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        let io_bandwidth = 64;
+        let (io_tx, io_rx) = channel(io_bandwidth);
+        let service = self
+            .service
+            .take()
+            .expect("same sink should not be run twice");
```

Comment:
`const`?

Reply:
I'm sorry, I don't understand the question.

Reply:
My bad. I wondered if making that string (same sink) into a `const` item would be worth doing. It might make `rustfmt` wrap the expression onto one line if you're careful.

Reply:
Aaah I see what you mean.